src/integration_tests/mavlink_ftp.cpp

```diff
@@ -251,3 +252,178 @@ TEST(MavlinkFTPTest, UploadFiles)
     }
 }
 
+TEST(MavlinkFTPTest, TestServer)
+{
+    ConnectionResult ret;
+    uint32_t crc;
+
+    Mavsdk dc_gcs;
+    dc_gcs.set_configuration(Mavsdk::Configuration::GroundStation);
+    ret = dc_gcs.add_udp_connection(24550);
+    ASSERT_EQ(ret, ConnectionResult::SUCCESS);
+    System &system_gcs = dc_gcs.system();
+
+    Mavsdk dc_cc;
+    dc_cc.set_configuration(Mavsdk::Configuration::CompanionComputer);
+    ret = dc_cc.setup_udp_remote("127.0.0.1", 24550);
+    ASSERT_EQ(ret, ConnectionResult::SUCCESS);
+    System &system_cc = dc_cc.system();
+
+    auto mavlink_ftp_server = std::make_shared<MavlinkFTP>(system_cc);
+    mavlink_ftp_server->set_root_dir("/usr/local");
+    uint8_t server_comp_id = mavlink_ftp_server->get_our_compid();
+
+    auto mavlink_ftp_client = std::make_shared<MavlinkFTP>(system_gcs);
+    mavlink_ftp_client->set_target_component_id(server_comp_id);
+    mavlink_ftp_client->set_timeout(10);
+    mavlink_ftp_client->set_retries(10);
+
+    {
```

Comment:
Functions would be nice, I'd say :).

Reply:
Looks so much better with functions. :-)